apps/emqx_conf/src/emqx_conf_app.erl

```diff
@@ -24,26 +24,60 @@
 -include_lib("emqx/include/logger.hrl").
 -include("emqx_conf.hrl").
 
+-define(DEFAULT_INIT_TXN_ID, -1).
+
 start(_StartType, _StartArgs) ->
     init_conf(),
     emqx_conf_sup:start_link().
 
 stop(_State) ->
     ok.
 
-%% internal functions
+get_override_config_file() ->
+    Node = node(),
+    Role = mria_rlog:role(),
+    case emqx_app:get_init_config_load_done() of
+        false ->
+            {error, #{node => Node, msg => "init_conf_load_not_done"}};
+        true when Role =:= core ->
```

Comment:
If business logic relies on the knowledge of the DB role, something smells fishy.

Reply:
I added a change with the second alternative: https://github.com/emqx/emqx/pull/7731/commits/67ed7ba7b8f7e0ce0437e9ad2d44e591c3ddf27b